cypress/page/personal-form.page.ts

```diff
@@ -27,71 +29,33 @@ class PersonalFormPage {
         this.modalTitle = "#example-modal-sizes-title-lg";
     }
 
-    public fillForm(personalInformation: { name: string; lastName: string; email: string; gender: string; dateOfBirth: string; mobileNumber: string; hobbies: string[]; currentAddress: string;}): void {
+    public visitPersonalInfoFormPage(): void{
+        cy.visit(this.personalInfoFormUrl);
+    }
+
+    public fillForm(personalInformation: PersonalFormPage.PersonalInformation): void {
         cy.get(this.nameInput).type(personalInformation.name);
         cy.get(this.lastnameInput).type(personalInformation.lastName);
         cy.get(this.emailInput).type(personalInformation.email);
         cy.get(this.genderRadioBtn).check(personalInformation.gender, {force: true});
         this.fillDateInput(personalInformation.dateOfBirth);
         cy.get(this.mobileNumberInput).type(personalInformation.mobileNumber);
-        personalInformation.hobbies.forEach(hobbie =>
-            {
-                cy.get(this.hobbiesCheckbox).filter(`:contains("${hobbie}")`).click();
-            });
+        personalInformation.hobbies.forEach(hobbie =>{
+            cy.get(this.hobbiesCheckbox).filter(`:contains("${hobbie}")`).click();
+        });
         cy.get(this.currentAddressInput).type(personalInformation.currentAddress);
         cy.get(this.submitBtn).click({force: true});
     }
 
     public fillDateInput(date: string){
-        let year;
-        let month;
-        let day;
+        const monthNames = ["January", "February", "March", "April", "May", "June",
+        "July", "August", "September", "October", "November", "December"
+        ];
```

Comment:
to avoid this logic, try to use the date.toLocaleString method. See one example here: https://bobbyhadz.com/blog/javascript-convert-month-number-to-name